### DISCUSSION TITLE
AHNet as 3D feature extractor

### DISCUSSION THREAD

**Discussion by joaolcguerreiro** [Project-MONAI/MONAI#6730] [General]
Hi,

is it possible to use [AHNet](https://docs.monai.io/en/stable/networks.html#ahnet) as a pre-trained 3D feature extractor?

In such case, what should be the transforms applied to the input tensors before feeding them to the pre-trained network? Should pixel values be rescaled? What were the training normalization statistics?

Also, which layers should be considered for feature extraction? Every layer except the last?

Thank you! :)

**Comment by joaolcguerreiro** [comment]
I have asked this question in [Model Zoo](https://github.com/Project-MONAI/model-zoo/discussions), since it may be more adequate :)

**Comment by KumoLiu** [comment]
Hi @joaolcguerreiro, thanks for your interest here!

> is it possible to use [AHNet](https://docs.monai.io/en/stable/networks.html#ahnet) as a pre-trained 3D feature extractor?

IMO any CNN can be used as a feature extractor.

> In such case, what should be the transforms applied to the input tensors before feeding them to the pre-trained network? Should pixel values be rescaled? What were the training normalization statistics?

For how to preprocess your data, I think it depends on your data instead of the network. For example, you can apply a Z-score normalization based on the mean and standard deviation of all collected intensity values for CT and for MRI or other image modalities simple Z-score normalization can be applied to the patient individually.

> Also, which layers should be considered for feature extraction? Every layer except the last?

IMO, yes. Each layer can be used as a feature extractor, except for the last layer which outputs logits.

Hope it helps, thanks!
